datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -632,16 +632,16 @@ impl LogicalPlan {
     /// params_values
     pub fn replace_params_with_values(
         &self,
-        param_values: &Vec<ScalarValue>,
+        param_values: &[ScalarValue],
     ) -> Result<LogicalPlan, DataFusionError> {
         let exprs = self.expressions();
-        let mut new_exprs = vec![];
+        let mut new_exprs = Vec::with_capacity(exprs.len());
         for expr in exprs {
             new_exprs.push(Self::replace_placeholders_with_values(expr, param_values)?);
         }
```

Comment:
I think you could make this slightly more idiomatic like this (untested):
```suggestion
        let new_exprs = exprs
          .into_iter()
          .map(|expr| Self::replace_placeholders_with_values(expr, param_values))
          .collect::<Result<Vec<_>>()?;
```

Also, same thing below

Reply:
Nice catching @alamb ! I've updated the code, please review.